.github/workflows/container-image.yml

```diff
@@ -38,6 +46,49 @@ jobs:
       id-token: write
 
     steps:
+      # Explicitly using the checkout action (instead of relying on docker/build-push-action to do it implicitly)
+      # because we need to fetch all tags.
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Switch to fetch-tags: true once https://github.com/actions/checkout/issues/1467 is fixed.
+          fetch-depth: 0
+
+      # Updates env.LATEST and env.LATEST_MAJOR based on
+      # whether the current release tag is the greatest overall and/or
+      # within its major version.
+      - name: Prepare metadata (release tags)
+        if: github.event_name == 'release' && github.event.action == 'published'
+        run: |
+          # Retrieve the greatest existing tag in the repository by sorting tags in descending order.
+          # Options used:
+          # * --sort=-v:refname sorts tags as versions, placing the highest version at the top.
+          # * -c 'versionsort.suffix=-' ensures that pre-release tags (e.g., 1.0.0-rc1) are sorted correctly,
+          #   so they are not considered greater than their corresponding final release (e.g., 1.0.0).
+          # Intentionally not using head -1 to prevent potential broken pipe errors.
+          greatest_tag=$(git -c 'versionsort.suffix=-' tag --list --sort=-v:refname | awk 'NR<=1')
```

Comment:
Even though I have written some awk, I failed to parse this at first glance. If you want to use awk for the mentioned reasons, why not say `NR==1`? Btw, there are different awk implementations and I am not sure if some will not optimize this query and terminate early.

Reply:
I just copied it from our GitLab actions. But yes, `NR==1` is better.

Reply:
Technically, everything is fine. I just wanted to comment on this line since it took me longer to grasp than the average shell script line should take. If we are using the same pattern somewhere else, this is okay for me.

Reply:
It's a valid point, and I've just adapted it.